### PR TITLE
Delete_post backend

### LINKED ISSUE #11: Create backend DELETE endpoint to delete posts
**Description**:  
Create backend DELETE endpoint to delete posts

### LINKED ISSUE #24: Post Management
**Description**:  
Create, delete and edit user post

**User Stories**:  
[Post Editing](https://github.com/itslupus/gamersnet/issues/23)
[Post Deletion](https://github.com/itslupus/gamersnet/issues/20)
[Post Creation](https://github.com/itslupus/gamersnet/issues/9)

### PR DESCRIPTION
**Pull request by jayKhakhariya**
This pull request closes #11 

> Backend point for deleting a single post
> Front part completed on different pr
> **EDIT:** Tests completed

### CODE REVIEW

**Comment by lucashuy** [comment]
Branch is currently stale

File: gamersnet_backend/persistence/posts.js [hunk 1]
```diff
@@ -54,5 +55,16 @@ async function addPost(userID, description, gameName, numPlayers, gameTimeUTC, d
   })
 }
 
+async function deletePost(_id, userID){
+  let db = await MongoDB.open();
+
+  let posts = db.collection('posts');
+
+  // just making sure the post is only deleted by the user that created it , just a note; for some reason (unknown)
+  // passing in userID was not working
```
Comment (humayraR) [inline gamersnet_backend/persistence/posts.js:77]: where did you want to pass in user id? the query?

File: gamersnet_backend/routes/posts/deletePost.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    let loggedIn = false;
+
+    if (cookie) {
+        cookie = cookie.split('=')[1];
+        loggedIn = await verifyUserLoggedIn(cookie);
+    }
+
+    if (loggedIn) {
+        let tokenDocument = await getUserIDFromToken(cookie);
+
```
Comment (humayraR) [inline gamersnet_backend/routes/posts/deletePost.js:23]: verify that the correct user(the owner) is deleting the post

**Review by humayraR — CHANGES_REQUESTED** [review]
Need to make sure the creator of the post is deleting it.

File: gamersnet_backend/routes/posts/deletePost.js (hunk 2 above)
Comment (jayKhakhariya) [reply gamersnet_backend/routes/posts/deletePost.js:23]: Yes, userID will be passed from here.

File: gamersnet_backend/persistence/posts.js (hunk 1 above)
Comment (jayKhakhariya) [reply gamersnet_backend/persistence/posts.js:77]: UserID is getting passed but for some reason(not important now) it was not working;

The new commit will only allow users that created the post to delete them.

**Review by jayKhakhariya — COMMENTED** [review]
> Branch is currently stale

I'll update it on my next commit (soon)

**Review by humayraR — DISMISSED** [review]

**Review by lucashuy — APPROVED** [review]
LGTM

**Pull request merged**